src/main/java/org/opentripplanner/middleware/aws/AwsApiGateway.java

```diff
@@ -0,0 +1,104 @@
+package org.opentripplanner.middleware.aws;
+
+import com.amazonaws.services.apigateway.AmazonApiGateway;
+import com.amazonaws.services.apigateway.AmazonApiGatewayClient;
+import com.amazonaws.services.apigateway.AmazonApiGatewayClientBuilder;
+import com.amazonaws.services.apigateway.model.CreateApiKeyRequest;
+import com.amazonaws.services.apigateway.model.CreateApiKeyResult;
+import com.amazonaws.services.apigateway.model.CreateUsagePlanKeyRequest;
+import com.amazonaws.services.apigateway.model.DeleteApiKeyRequest;
+import com.amazonaws.services.apigateway.model.GetUsagePlanRequest;
+import com.amazonaws.services.apigateway.model.GetUsagePlanResult;
+import com.amazonaws.services.apigateway.model.NotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Manages all interactions with AWS api gateway.
+ */
+public class AwsApiGateway {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AwsApiGateway.class);
+    private static final int SDK_REQUEST_TIMEOUT = 10 * 1000;
+    private static final String AWS_API_GATEWAY_USAGE_PLAN_ID
+        = getConfigPropertyAsText("AWS_API_GATEWAY_USAGE_PLAN_ID");
+
+    /**
+     * Create connection to AWS api gateway.
+     */
+    private static AmazonApiGateway getAmazonApiGateway() {
+        AmazonApiGatewayClientBuilder gatewayBuilder = AmazonApiGatewayClient.builder();
+        AmazonApiGateway gateway = gatewayBuilder.build();
+        return gateway;
+    }
+
+    /**
+     * Request an API key from AWS api gateway and assign it to an existing usage plan.
+     */
+    public static String getApiKeyId(String userId) {
+        long startTime = System.currentTimeMillis();
+
+        try {
+
+            AmazonApiGateway gateway = getAmazonApiGateway();
+
+            // create API key
+            CreateApiKeyRequest apiKeyRequest = new CreateApiKeyRequest();
+            apiKeyRequest.setSdkRequestTimeout(SDK_REQUEST_TIMEOUT);
+            apiKeyRequest
+                .withName(userId)
+                .withCustomerId(userId)
+                .withEnabled(true);
+            CreateApiKeyResult apiKeyResult = gateway.createApiKey(apiKeyRequest);
+
+            // get existing usage plan
+            GetUsagePlanRequest usagePlanRequest = new GetUsagePlanRequest();
+            usagePlanRequest.withUsagePlanId(AWS_API_GATEWAY_USAGE_PLAN_ID);
```

Comment:
I think perhaps this method should take a Usage Plan Id as a param for now and perhaps in the ApiUserController preCreateHook invocation, we can pass the DEFAULT_USAGE_PLAN_ID (which is derived from the config). I think we'll want to be able to manage/create keys for multiple usage plans, though how exactly that will work is not defined yet.

Reply:
Updated.